flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerConfiguration.java

```diff
@@ -259,6 +269,7 @@ public static TaskManagerConfiguration fromConfiguration(Configuration configura
 
 		return new TaskManagerConfiguration(
 			numberSlots,
+			taskExecutorResourceSpec.getDefaultSlotResourceProfile(),
```

Comment:
I think re-computation of `TaskExecutorResourceSpec` takes negligible time.
For cleanliness, I would consider using here suggested before `TaskExecutorResourceUtils#generateDefaultSlotProfile(config)`.
Then we do not need changes in TM config classes.

Reply:
If we do not change `TaskManagerConfiguration`, we will have to derive `TaskExecutorResourceSpec` and generate default slot resource profile again in `TaskExecutor#connectToResourceManager`.
I would avoid such calculations after `TaskExecutor` being launched.

Reply:
I meant not sharing `TaskExecutorResourceSpec` between `TaskManagerServicesConfiguration` and `TaskManagerConfiguration` but calculating them inside `fromConfiguration` to avoid changes.

Reply:
As discussed offline, I'll derive default slot resource profile outside of `TaskManagerConfiguration`, but keep reusing the same `TaskExecutorResourceSpec` in `TaskManagerServiceConfiguration` and for deriving default slot resource profile for `TaskManagerConfiguration`, to avoid the chance of inconsistency.